### PR TITLE
[TTS] Remove un-supported tts models and related docs

### PR DESCRIPTION
**Pull request by subhankar-ghosh**
> [!IMPORTANT]  
> The `Update branch` button must only be pressed in very rare occassions.
> An outdated branch is never blocking the merge of a PR.
> Please reach out to the automation team before pressing that button.

# What does this PR do ?

Remove support for the following models from TTS collection:
- MixerTTS
- RadTTS
- tacotron2
- VITS
- Waveglow
- Univnet
- Spectrogram Enhancer

Remove code, remove tests, docs, configs

**Collection**: [Note which collection this PR will affect]

TTS

# Usage

- You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

**Comment by subhankar-ghosh** [comment]
@rlangman @paarthneekhara @shehzeen @rfejgin @artem-gorodetskii Please check if you are using any of the files/methods/classes being removed.

**Comment by pzelasko** [comment]
Please fix the import error in `test_hybrid_asr_tts_models.py`

**Review by blisc — CHANGES_REQUESTED** [review]
Blocking until I get a chance to review

File: docs/source/tts/data/ngc_models_aligner.csv [hunk 1]
```diff
@@ -1,3 +1 @@
 Locale,Model Name,Dataset,Sampling Rate,#Spk,Phoneme Unit,Model Class,Overview,Checkpoint
```
Comment (blisc) [inline docs/source/tts/data/ngc_models_aligner.csv]: Since this file is now empty, can we remove it?

File: docs/source/tts/data/ngc_models_e2e.csv [hunk 2]
```diff
@@ -1,3 +1 @@
 Locale,Model Name,Dataset,Sampling Rate,#Spk,Phoneme Unit,Model Class,Overview,Checkpoint
```
Comment (blisc) [inline docs/source/tts/data/ngc_models_e2e.csv]: Since this file is now empty, can we remove it?

File: docs/source/tts/api.rst
Comment (blisc) [inline docs/source/tts/api.rst:1]: We need to add magpie into this somewhere

File: examples/tts/test_tts_infer.py [hunk 3]
```diff
@@ -57,13 +57,13 @@ def main():
     parser.add_argument(
```
Comment (blisc) [inline examples/tts/test_tts_infer.py]: Do we still want this file? Maybe we should just keep two: one for magpie and one for fastpitch

File: nemo/collections/tts/data/dataset.py [hunk 4]
```diff
@@ -1161,43 +1055,6 @@ def __len__(self):
...
-        lengths = [spec.shape[0] for spec in true_specs]
-
-        pred_specs = torch.nn.utils.rnn.pad_sequence(pred_specs, batch_first=True)
-        true_specs = torch.nn.utils.rnn.pad_sequence(true_specs, batch_first=True)
-        lengths = torch.LongTensor(lengths)
-
-        return rearrange(pred_specs, "b l c -> b c l"), rearrange(true_specs, "b l c -> b c l"), lengths
-
-
 class FastPitchSSLDataset(Dataset):
```
Comment (blisc) [inline nemo/collections/tts/data/dataset.py:1058]: Do we want to keep FastPitch SSL?

File: nemo/collections/tts/g2p/models/t5.py
Comment (blisc) [inline nemo/collections/tts/g2p/models/t5.py:14]: Maybe we should remove this too

File: nemo/collections/tts/README.md [hunk 5]
```diff
@@ -1,7 +1,7 @@
 # Text-to-Speech Synthesis (TTS):
 
 * [Documentation](https://docs.nvidia.com/deeplearning/nemo/user-guide/docs/en/main/tts/intro.html#)
-* Mel-Spectrogram generators: FastPitch, SSL FastPitch, Mixer-TTS/Mixer-TTS-X, RAD-TTS, Tacotron2
-* Vocoders: HiFiGAN, UnivNet, WaveGlow
-* End-to-End Models: VITS
+* Mel-Spectrogram generators: FastPitch, SSL FastPitch
+* Vocoders: HiFiGAN
+* Spectrogram enhancer for ASR fine-tuning
```
Comment (blisc) [inline nemo/collections/tts/README.md:6]: I think this was removed

File: tests/functional_tests/L2_TTS_Fast_dev_runs_1_Tacotron_2.sh [hunk 6]
```diff
@@ -1,31 +0,0 @@
-# Copyright (c) 2020-2025, NVIDIA CORPORATION.
```
Comment (blisc) [inline tests/functional_tests/L2_TTS_Fast_dev_runs_1_Tacotron_2.sh:1]: I think there's another test file that you need to modify to tell CI not to run these anymore. Some sort of github actions yaml

**Review by blisc — APPROVED** [review]
Most of the PR is good. we can merge this one first and do further pruning in another PR

File: nemo/collections/tts/data/dataset.py (hunk 4 above)
Comment (subhankar-ghosh) [reply nemo/collections/tts/data/dataset.py:1058]: @rlangman Do we need to keep FastPitch SSL? I remember you worked on Fastpitch working on audio codecs, are these related?

Comment (rlangman) [reply nemo/collections/tts/data/dataset.py:1058]: I don't think we need Fastpitch SSL. It is not related to any of the audio codec work.

File: tests/functional_tests/L2_TTS_Fast_dev_runs_1_Tacotron_2.sh (hunk 6 above)
Comment (subhankar-ghosh) [reply tests/functional_tests/L2_TTS_Fast_dev_runs_1_Tacotron_2.sh:1]: Done

File: nemo/collections/tts/README.md (hunk 5 above)
Comment (subhankar-ghosh) [reply nemo/collections/tts/README.md:6]: Done

File: examples/tts/test_tts_infer.py (hunk 3 above)
Comment (subhankar-ghosh) [reply examples/tts/test_tts_infer.py]: Removed this file.

File: docs/source/tts/api.rst
Comment (subhankar-ghosh) [reply docs/source/tts/api.rst:1]: Done

File: docs/source/tts/data/ngc_models_aligner.csv (hunk 1 above)
Comment (subhankar-ghosh) [reply docs/source/tts/data/ngc_models_aligner.csv]: Done

File: docs/source/tts/data/ngc_models_e2e.csv (hunk 2 above)
Comment (subhankar-ghosh) [reply docs/source/tts/data/ngc_models_e2e.csv]: Done
